### PR TITLE
success and failure tests for large events

### PR DESCRIPTION
**Pull request by stormeye2000**

### CODE REVIEW

File: src/main/java/com/stormeye/producer/service/producer/ProducerService.java [hunk 1]
```diff
@@ -69,7 +69,13 @@ void sendEvent(final URI emitter, final Event<?> event) {
 
         logger.debug("Emitter: [{}] Topic: [{}] - Event : [{}]", emitter, topic, event);
 
-        kafkaProducer.send(new ProducerRecord<>(topic, 0, System.currentTimeMillis(), null, event), new SendCallback());
+        kafkaProducer.send(new ProducerRecord<>(topic, 0, System.currentTimeMillis(), null, event), (metadata, exception) -> {
+            if (exception != null) {
+                logger.error("Error producing event - Metadata: [{}]\r\n{}", metadata, exception.getMessage());
```
Comment (meywood) [inline src/main/java/com/stormeye/producer/service/producer/ProducerService.java]: Don't pass in the message pass in the whole exception:
`logger.error("Error producing event - Metadata: [{}]", metadata, exception);`

File: src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventFail.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import com.casper.sdk.model.event.Event;
+import com.stormeye.producer.config.AppConfig;
+import com.stormeye.producer.config.ServiceProperties;
+
+import java.util.concurrent.Future;
+
+@SpringBootTest(classes = {AppConfig.class, ServiceProperties.class})
```
Comment (meywood) [inline src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventFail.java:18]: JavaDoc on class explaining what your testing

File: src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventSuccess.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import com.casper.sdk.model.event.Event;
+import com.stormeye.producer.config.AppConfig;
+import com.stormeye.producer.config.ServiceProperties;
+
+import java.util.concurrent.Future;
+
+@SpringBootTest(classes = {AppConfig.class, ServiceProperties.class})
```
Comment (meywood) [inline src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventSuccess.java:20]: JavaDoc explaining what this class is testing

File: src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventSuccess.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        final Future<RecordMetadata> send = kafkaProducer.send(producerRecord, null);
+
+        while (!send.isDone()){
+            Thread.sleep(1000);
+        }
+
+        try {
+            send.get();
+            assertThat(Boolean.TRUE, is(Boolean.TRUE));
```
Comment (meywood) [inline src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventSuccess.java]: These asserts do nothing

File: src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventFail.java (hunk 2 above)
Comment (cnorburn) [reply src/test/java/com/stormeye/producer/service/producer/send/SendMegaEventFail.java:18]: done

**Review by meywood — APPROVED** [review]

**Pull request merged**